### PR TITLE
Fix three coverity issues

### PR DESCRIPTION
**Pull request by paulidale**
Fixing Coverity issues 1507372, 1503321 & 1503327.

### CODE REVIEW

**Review by beldmit — APPROVED** [review]
LGTM

File: test/evp_extra_test2.c [hunk 1]
```diff
@@ -346,10 +346,12 @@ static int test_dh_tofrom_data_select(void)
           && TEST_int_gt(EVP_PKEY_generate(gctx, &key), 0)
           && TEST_true(do_pkey_tofrom_data_select(key, "DHX"));
 # ifndef OPENSSL_NO_DEPRECATED_3_0
-    dhkey = EVP_PKEY_get0_DH(key);
-    ret = ret && TEST_ptr(dhkey);
-    ret = ret && TEST_ptr(privkey = DH_get0_priv_key(dhkey))
-              && TEST_int_le(BN_num_bits(privkey), 225);
+    if (key != NULL) {
+        dhkey = EVP_PKEY_get0_DH(key);
+        ret = ret && TEST_ptr(dhkey)
```
Comment (t8m) [inline test/evp_extra_test2.c:351]: IMO putting the dhkey assignment into this TEST_ptr() call would work too and be more concise. But OK as is.

**Review by t8m — APPROVED** [review]

File: test/evp_extra_test2.c (hunk 1 above)
Comment (paulidale) [reply test/evp_extra_test2.c:351]: There will be an error message if `key == NULL`, so a TEST_ptr call is kind of irrelevant.

Comment (t8m) [reply test/evp_extra_test2.c:351]: There is the `ret = ret &&` so you could drop the `if (key != NULL)` around these calls.

Comment (paulidale) [reply test/evp_extra_test2.c:351]: Two error messages could confuse.

Comment (t8m) [reply test/evp_extra_test2.c:351]: There would not be as the ret would be false already.

Comment (t8m) [reply test/evp_extra_test2.c:351]: `ret = ret && TEST_ptr(dhkey = EVP_PKEY_get0_DH(key));`

Comment (paulidale) [reply test/evp_extra_test2.c]: Okay, this makes sense.

I'm in a world of my own today :(

**Comment by paulidale** [comment]
Merged, thanks for the reviews.

**Pull request closed without merging**